pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -57,6 +57,9 @@ class Hadamard(Observable, Operation):
 
     _queue_category = "_ops"
 
+    def __init__(self, wires: Optional[WiresLike] = None, id: Optional[str] = None):
```

Comment:
This is such that an IDE does not complain about the `0` in `qml.H(0)` not being "`TensorLike`"

Reply:
```suggestion
    def __init__(self, wires: WiresLike, id: Optional[str] = None):
```
Though the issue with built-ins and `TensorLike` is a deeper issue we need to address at some point.